src/test/java/com/jcabi/jdbc/JdbcSessionITCase.java

```diff
@@ -91,6 +98,84 @@ public void changesTransactionIsolationLevel() throws Exception {
     }
 
     /**
+     * JdbcSession can run a function (stored procedure) with
+     * output parameters.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void callsFunctionWithOutParam() throws Exception {
+        final DataSource dsrc = JdbcSessionITCase.source();
+        new JdbcSession(dsrc).autocommit(false).sql(
+            "CREATE TABLE IF NOT EXISTS users (name VARCHAR(50))"
+        ).execute().sql("INSERT INTO users (name) VALUES (?)")
+        .set("Jeff Charles").execute().sql(
+            StringUtils.join(
+                "CREATE OR REPLACE FUNCTION fetchUser(username OUT text)",
+                " AS $$ BEGIN SELECT name INTO username from  users; END;",
+                " $$ LANGUAGE plpgsql;"
+            )
+        ).execute().commit();
+        final Object[] result = new JdbcSession(dsrc)
+            .sql("{call fetchUser(?)}")
+            .prepare(
+                new Preparation() {
+                    @Override
+                    public void prepare(
+                        final PreparedStatement stmt) throws SQLException {
+                        ((CallableStatement) stmt)
```

Comment:
@mkordas In fact, I don't even know how to get rid of it; make a redesign and change stuff, add flow just for ``CallableStatement``, all just to avoid a type cast...